### PR TITLE
kubelet/eviction: improve unit test coverage

### PR DESCRIPTION
**Pull request by tmilox**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will improve the unit test coverage of `pkg/kubelet/eviction`.

- before

Screenshot 2024-05-21 at 07 47 16

- after

Screenshot 2024-05-22 at 17 15 02

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by TommyStarK** [comment]
I'm trying to get back into contributing, so to get back on track I thought I'd start by improving unit test code coverage. I apologize in advance if some of the tests/cases added are irrelevant. I lack a lot of knowledge in this area. I hope it will be useful anyway.

File: pkg/kubelet/eviction/helpers_test.go [hunk 1]
```diff
@@ -110,6 +110,22 @@ func TestGetReclaimableThreshold(t *testing.T) {
 			t.Errorf("Didn't find reclaimable threshold, test: %v", testName)
 		}
 	}
+	cornerCases := map[string]struct {
```
Comment (bart0sh) [inline pkg/kubelet/eviction/helpers_test.go:113]: Can this case be added to the `testCases` instead?

File: pkg/kubelet/eviction/helpers_test.go [hunk 2]
```diff
@@ -648,6 +736,50 @@ func TestAddAllocatableThresholds(t *testing.T) {
...
+			usageFunc: func() int64 {
+				return int64(processUsage(&statsapi.ProcessStats{}))
+			},
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			value := test.usageFunc()
+			if value != 0 {
+				t.Errorf("expected '%s' to have a fallback value equal to 0, but got %d", test.usageFuncName, value)
+			}
```
Comment (bart0sh) [inline pkg/kubelet/eviction/helpers_test.go:778]:
```suggestion
                         assert.NotEqual(t, test.usageFunc(), 0, "unexpected fallback value")
```

File: pkg/kubelet/eviction/helpers_test.go [hunk 3]
```diff
@@ -3250,3 +3382,52 @@ func TestEvictonMessageWithResourceResize(t *testing.T) {
...
+			compFunc: func(stats statsFunc) cmpFunc {
+				return disk(stats, nil, "")
+			},
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			result := test.compFunc(statsFn)(pod1, pod2)
+			if result != 0 {
+				t.Errorf("expected 0 but got %d", result)
+			}
```
Comment (bart0sh) [inline pkg/kubelet/eviction/helpers_test.go:3430]: The same here, please use [testify](https://github.com/stretchr/testify) assert/require APIs.

File: pkg/kubelet/eviction/helpers_test.go (hunk 2 above)
Comment (TommyStarK) [reply pkg/kubelet/eviction/helpers_test.go]: resolved in e732e7535ae

File: pkg/kubelet/eviction/helpers_test.go (hunk 1 above)
Comment (TommyStarK) [reply pkg/kubelet/eviction/helpers_test.go]: resolved in e732e7535ae

**Comment by TommyStarK** [comment]
Thanks for the review @bart0sh, I've addressed your comments.

File: pkg/kubelet/eviction/helpers_test.go [hunk 4]
```diff
@@ -648,6 +733,47 @@ func TestAddAllocatableThresholds(t *testing.T) {
...
+		{
+			description:   "process usage, fallback value",
+			usageFuncName: "processUsage",
+			usageFunc: func() int64 {
+				return int64(processUsage(&statsapi.ProcessStats{}))
+			},
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			assert.NotEqual(t, 0, test.usageFunc(), "unexpected fallback value")
```
Comment (bart0sh) [inline pkg/kubelet/eviction/helpers_test.go:772]: I don't see how `test.usageFuncName` is used in the test. It seems that tests only calls `test.usageFunc` and expects it to return non-zero result.

Comment (TommyStarK) [reply pkg/kubelet/eviction/helpers_test.go]: my bad, resolved in 4632fc8a0f3

File: pkg/kubelet/eviction/helpers_test.go [hunk 5]
```diff
@@ -102,12 +104,23 @@ func TestGetReclaimableThreshold(t *testing.T) {
...
 	}
 	for testName, testCase := range testCases {
 		sort.Sort(byEvictionPriority(testCase.thresholds))
 		_, _, ok := getReclaimableThreshold(testCase.thresholds)
-		if !ok {
+		if !ok && !testCase.shouldNotHaveFound {
 			t.Errorf("Didn't find reclaimable threshold, test: %v", testName)
+		} else if ok && testCase.shouldNotHaveFound {
+			t.Errorf("Shouldn't have found reclaimable threshold, test: %v", testName)
 		}
```
Comment (bart0sh) [inline pkg/kubelet/eviction/helpers_test.go:124]: It's not very clear what will happen in other cases, e.g. when ok is True, but shouldNotHavefound is false. Can you cleanup the code a little bit?

Comment (TommyStarK) [reply pkg/kubelet/eviction/helpers_test.go]: resolved in 47a74aebe8b

File: pkg/kubelet/eviction/memory_threshold_notifier_test.go [hunk 6]
```diff
@@ -152,6 +152,48 @@ func TestUpdateThreshold(t *testing.T) {
...
+	}
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			m := newTestMemoryThresholdNotifier(evictionapi.Threshold{}, nil, nil)
+			if tc.allocatableEvictionThreshold {
+				m.threshold.Signal = evictionapi.SignalAllocatableMemoryAvailable
+			}
+			if err := m.UpdateThreshold(tc.summary); err == nil {
+				t.Errorf("Expected error updating threshold, but got nil")
+			}
```
[the remark this answers is not in the document]
Comment (TommyStarK) [reply pkg/kubelet/eviction/memory_threshold_notifier_test.go]: resolved in 47a74aebe8b

**Comment by bart0sh** [comment]
/lgtm

/assign @SergeyKanzhelev @mrunalp 
for approval

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**